### PR TITLE
allows booleans for stream default values

### LINKED ISSUE #304: Allow boolean stream targets in JOI schemas of the device
We are about to intrroduce boolean streams with values of `true` and `false`.
Schemas which limit this values to numbers need to be updated!

### PR DESCRIPTION
**Pull request by vdiezel**
closes #304 
Allows user defined stream default to also be of type `boolean`. 
shipped to dev and prod.

### CODE REVIEW

File: services/device/src/joi/device.js [hunk 1]
```diff
@@ -108,7 +108,7 @@ const setStreamDefault = createSchemaAssertion(Joi.object({
   deviceId: Joi.string().required().pattern(UUIDV4_REGEX),
   data: Joi.object({
     streamId: Joi.string().required(),
-    value: Joi.number().required()
+    value: Joi.alternatives().try(Joi.number(), Joi.boolean()).required()
```
Comment (DanielSeehausen) [inline services/device/src/joi/device.js]: was trying to look up if required worked on alternatives, as this is new JOI territory for me
and came across this VERY juicy thread: https://github.com/hapijs/joi/issues/1713

oh many, this one is also crazy with a good twist!:
https://github.com/hapijs/joi/issues/1304

**Comment by DanielSeehausen** [comment]
updated branch as it was clean rebase for tests

File: services/device/src/joi/device.js (hunk 1 above)
Comment (vdiezel) [reply services/device/src/joi/device.js]: Marsup - you gotta love that guy

File: services/device/test/device/device.test.js [hunk 2]
```diff
@@ -119,43 +119,53 @@ describe('Device Class', () => {
...
 
-    test('shoudl set a streamDefault value when a streamId that exists on the device is used', async() => {
-      const stream = generateStream()
-      const { user, device } = await seedUserDevice({ streams: [stream] }, ['USER_AS_ADMIN'])
+    test('should set a streamDefault value when a streamId that exists on the device is used', async() => {
 
-      expect(device.streamDefaults.length).toBe(0)
+      const streams = Array.from({ length: 2 }, () => generateStream())
+      const streamDefaults = [
+        { value: false, streamId: streams[0].id },
```
Comment (ConorTighe1995) [inline services/device/test/device/device.test.js]: Does stream [0] and [1] hold and kind of significance? Like downstream/upstream?

**Pull request merged**